Makefile.features

```diff
@@ -1,3 +1,33 @@
 # Process FEATURES variables
 
 include $(RIOTBOARD)/$(BOARD)/Makefile.features
+
+
+# Resolve FEATURES_ variables
+# Their value will only be complete after resolving dependencies
+
+# Features that are required by the application but not provided by the BSP
+# Having features missing may case the build to fail.
+FEATURES_MISSING = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_REQUIRED)))
+
+# Features that are only optional and not required at the same time.
+# The policy is to use by default whenever they are provided by the BSP.
+FEATURES_OPTIONAL_ONLY = $(sort $(filter-out $(FEATURES_REQUIRED),$(FEATURES_OPTIONAL)))
+FEATURES_OPTIONAL_USED = $(sort $(filter $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
+# Optional features that will not be used because they are unavailable
+FEATURES_OPTIONAL_UNUSED = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
```

Comment:
Thank you for going in the semantic details, that is what I want to be challenged :) 

I wanted to separate `missing` which for me sounds like an error from `they are not available but optional anyway/or not used`. And even changed it from the previous `FEATURES_MISSING` message that was in `info-build`.

https://github.com/RIOT-OS/RIOT/pull/11492/files#diff-cb9c0ad59ebaddd68b4a35e09c6d47f5R56

With a new general `FEATURES_BLACKLIST` https://github.com/RIOT-OS/RIOT/pull/9081 variable or specific `FEATURES_OPTIONAL_DISABLE`  https://github.com/RIOT-OS/RIOT/issues/10047#issuecomment-425437023 variable, an optional feature could be provided but not included.

But maybe I would indeed need a specific variable for this case or the `OPTIONAL_MISSING`.

> About the first two commits, do you see more use for the recipe besides testing this PR? If there is one lets keep it, but if not remove them from the PR.

Testing other refactoring Pull requests that touch the makefiles/info-global.inc.mk` and can change `info-boards-supported`.
I would prefer to have them in a separate PR anyway.

Reply:
> I wanted to separate missing which for me sounds like an error from they are not available but optional anyway/or not used. And even changed it from the previous FEATURES_MISSING message that was in info-build.

I agree with this. But FEATURES_OPTIONAL_UNUSED makes me think of features that are optional but that I (or the build system) have decided not to use. I had to read up the comment and code to understand what it meant since it went against my intuition of what the variable name made me think.

> FEATURES_OPTIONAL_DISABLE #10047 (comment) variable, an optional feature could be provided but not included.

In that case to me its more intuitive to have:

```
FEATURES_OPTIONAL_UNUSED = $(sort $(filter $(FEATURES_OPTIONAL_DISABLE),$(FEATURES_OPTIONAL_ONLY)))
```
and

```
FEATURES_OPTIONAL_MISSING = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))

```

Anyway this is just my interpretation, but I think is exactly the kind of questioning you wanted right?

Reply:
So you would prefer two variables for this.
Currently there would be no `FEATURES_OPTIONAL_UNUSED` as none of the two `DISABLE` or `BLACKLIST` are there.

I will change the `UNUSED` to go back to the `MISSING` behavior and see for a nice more describing comment.